lua/lint/linters/swiftlint.lua

```diff
@@ -0,0 +1,75 @@
+local pattern = "[^:]+:(%d+):(%d+): (%w+): (.+)"
+local groups = { "lnum", "col", "severity", "message" }
+local defaults = { ["source"] = "swiftlint" }
+local severity_map = {
+  ["error"] = vim.diagnostic.severity.ERROR,
+  ["warning"] = vim.diagnostic.severity.WARN,
+}
+
+local cached_config = nil
+local searched_for_config = false
+
+local function find_config(filename)
+  if searched_for_config then
+    return cached_config
+  end
+
+  local configs = vim.fn.systemlist({
```

Comment:
`vim.fs.find` is quite inefficient, because you can't provide maximum depth. Therefore, using downward direction it will traverse all directories. If someone opens a swift file in home dir it will cause a quite bad effect.

Isn't find built-in both into Linux and macOS?

Reply:
I wanted to cover a case when people keep config file a nested directory. Otherwise, upward traverse would be enough.

Reply:
> vim.fs.find is quite inefficient, because you can't provide maximum depth. Therefore, using downward direction it will traverse all directories.

Then you could use `vim.fs.dir`, it takes a depth and skip argument.

> Isn't find built-in both into Linux and macOS?

It's not built into the linux kernel. I don't know if every distribution ships it by default. Linux distributions and MacOS are not the only systems that exist.

Reply:
> Then you could use vim.fs.dir, it takes a depth and skip argument.
Cool, I didn't know about `vim.fs.dir`. I'll try to replace `find` with it.

Reply:
I started thinking about it and I decided that it's better to keep it simple and avoid handling some very project-specific cases. 

I don't see any reason why the config file couldn't be placed in the project root dir. Therefore, I switched to `vim.fs.dir` with `upward` traverse.